EOLib/Domain/Character/WalkValidationActions.cs

```diff
@@ -16,6 +24,10 @@ public class WalkValidationActions : IWalkValidationActions
         private readonly ICurrentMapStateProvider _currentMapStateProvider;
         private readonly IUnlockDoorValidator _unlockDoorValidator;
 
+        private DateTime? _startedGhosting = null;
+        private Character _currentlyGhosting = null;
```

Comment:
One thing I'm not big on is storing state in this class directly. My ideal goal was to have state represented by repository/provider(s) throughout the code base, but there are a number of services that also have their own private state (see: `Characteranimator`). With the `Actions` layer though I would like to avoid it; I think it makes sense to introduce a repository for these objects.

Reply:
This has now been moved into a `GhostingRepository`